### ISSUE TITLE
`IndexError` when using Jinja for-loops with dbt config

### ISSUE THREAD

**Issue by SoheilSalmani** [sqlfluff/sqlfluff#1435]
## Expected Behaviour

No error.

## Observed Behaviour

```
❯ sqlfluff lint
=== [dbt templater] Compiling dbt project...
=== [dbt templater] Project Compiled.
WARNING    Unable to lint /workspaces/python/bug_sqlfluff/my_dbt_project/models/example/sales.sql due to an internal error. Please report this as an issue with your query's contents and stacktrace below!
To hide this warning, add the failing file to .sqlfluffignore
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/sqlfluff/core/linter/runner.py", line 95, in run
    yield partial()
  File "/usr/local/lib/python3.9/site-packages/sqlfluff/core/linter/linter.py", line 512, in lint_rendered
    parsed = cls.parse_rendered(rendered)
  File "/usr/local/lib/python3.9/site-packages/sqlfluff/core/linter/linter.py", line 294, in parse_rendered
    tokens, lvs, config = cls._lex_templated_file(
  File "/usr/local/lib/python3.9/site-packages/sqlfluff/core/linter/linter.py", line 127, in _lex_templated_file
    tokens, lex_vs = lexer.lex(templated_file)
  File "/usr/local/lib/python3.9/site-packages/sqlfluff/core/parser/lexer.py", line 319, in lex
    segments: Tuple[RawSegment, ...] = self.elements_to_segments(
  File "/usr/local/lib/python3.9/site-packages/sqlfluff/core/parser/lexer.py", line 346, in elements_to_segments
    source_slice = templated_file.templated_slice_to_source_slice(
  File "/usr/local/lib/python3.9/site-packages/sqlfluff/core/templaters/base.py", line 289, in templated_slice_to_source_slice
    start_slices = [self.sliced_file[ts_start_sf_start]]
IndexError: list index out of range
 
All Finished 📜 🎉!
```

```
❯ sqlfluff parse my_dbt_project/models/example/sales.sql 
=== [dbt templater] Compiling dbt project...
=== [dbt templater] Project Compiled.
Traceback (most recent call last):
  File "/usr/local/bin/sqlfluff", line 8, in <module>
    sys.exit(cli())
  File "/usr/local/lib/python3.9/site-packages/click/core.py", line 1137, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/click/core.py", line 1062, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python3.9/site-packages/click/core.py", line 1668, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python3.9/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python3.9/site-packages/click/core.py", line 763, in invoke
    return __callback(*args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/sqlfluff/cli/commands.py", line 713, in parse
    for parsed_string in result:
  File "/usr/local/lib/python3.9/site-packages/sqlfluff/core/linter/linter.py", line 860, in parse_path
    yield self.parse_string(
  File "/usr/local/lib/python3.9/site-packages/sqlfluff/core/linter/linter.py", line 596, in parse_string
    return self.parse_rendered(rendered, recurse=recurse)
  File "/usr/local/lib/python3.9/site-packages/sqlfluff/core/linter/linter.py", line 294, in parse_rendered
    tokens, lvs, config = cls._lex_templated_file(
  File "/usr/local/lib/python3.9/site-packages/sqlfluff/core/linter/linter.py", line 127, in _lex_templated_file
    tokens, lex_vs = lexer.lex(templated_file)
  File "/usr/local/lib/python3.9/site-packages/sqlfluff/core/parser/lexer.py", line 319, in lex
    segments: Tuple[RawSegment, ...] = self.elements_to_segments(
  File "/usr/local/lib/python3.9/site-packages/sqlfluff/core/parser/lexer.py", line 346, in elements_to_segments
    source_slice = templated_file.templated_slice_to_source_slice(
  File "/usr/local/lib/python3.9/site-packages/sqlfluff/core/templaters/base.py", line 289, in templated_slice_to_source_slice
    start_slices = [self.sliced_file[ts_start_sf_start]]
IndexError: list index out of range
```

## Steps to Reproduce

```bash
pip install sqlfluff==0.6.6 dbt==0.20.2
```

```sql
-- models/example/sales.sql

SELECT
    date_key,
    store_id,
    units_sold,
  {%- for offset in range(1, 5) %}
    LAG(units_sold, {{ offset }}) OVER (
        PARTITION BY store_id, EXTRACT(DAYOFWEEK FROM date_key)
        ORDER BY date_key
    ) AS units_sold_{{ offset }}_weeks_ago{% if not loop.last %},{% endif -%}
  {% endfor %}

FROM `project_id`.`dataset_id`.`table_id`
```

Compiled version generated by dbt:

```sql
--  target/compiled/.../models/example/sales.sql

SELECT
    date_key,
    store_id,
    units_sold,
    LAG(units_sold, 1) OVER (
        PARTITION BY store_id, EXTRACT(DAYOFWEEK FROM date_key)
        ORDER BY date_key
    ) AS units_sold_1_weeks_ago,
    LAG(units_sold, 2) OVER (
        PARTITION BY store_id, EXTRACT(DAYOFWEEK FROM date_key)
        ORDER BY date_key
    ) AS units_sold_2_weeks_ago,
    LAG(units_sold, 3) OVER (
        PARTITION BY store_id, EXTRACT(DAYOFWEEK FROM date_key)
        ORDER BY date_key
    ) AS units_sold_3_weeks_ago,
    LAG(units_sold, 4) OVER (
        PARTITION BY store_id, EXTRACT(DAYOFWEEK FROM date_key)
        ORDER BY date_key
    ) AS units_sold_4_weeks_ago

FROM `project_id`.`dataset_id`.`table_id`
```

## Dialect
BigQuery

## Version
sqlfluff, version 0.6.6

## Configuration
```
[sqlfluff]
templater = dbt
dialect = bigquery

[sqlfluff:templater:dbt]
project_dir = my_dbt_project/
profiles_dir = my_dbt_project/secrets/dbt_profiles/
profile = default
```

**Comment by tunetheweb** [comment]
This looks like a duplicate of #1414 which should be fixed by #1430 which is about to be merged.

**Comment by barrywhart** [comment]
Confirmed fixed in `main` as of commit `9eb30226d77727cd613947e144a0abe483151f18`.
